extensions/measurement-tracking/src/contexts/TrackedMeasurementsContext/TrackedMeasurementsContext.jsx

```diff
@@ -58,6 +122,42 @@ function TrackedMeasurementsContextProvider(
     trackedMeasurementsService,
   ] = useMachine(measurementTrackingMachine);
 
+  // ~~ Listen for changes to ViewportGrid for potential SRs hung in panes when idle
+  useEffect(() => {
+    if (viewports.length > 0) {
+      const activeViewport = viewports[activeViewportIndex];
+
+      if (!activeViewport || !activeViewport.displaySetInstanceUID) {
+        return;
+      }
+
+      const { DisplaySetService } = servicesManager.services;
+      const displaySet = DisplaySetService.getDisplaySetByUID(
+        activeViewport.displaySetInstanceUID
+      );
+
+      // Magic string
+      // load function added by our sopClassHandler module
+      if (
+        displaySet.SOPClassHandlerId ===
+          'org.ohif.dicom-sr.sopClassHandlerModule.dicom-sr' &&
+        !displaySet.isLocked
+      ) {
+        console.log('sending event...', trackedMeasurements);
+        sendTrackedMeasurementsEvent('PROMPT_HYDRATE_SR', {
+          // Why not just pass the displaySet ? We will end up searching for it again to call load.
```

Comment:
> // Why not just pass the displaySet ? We will end up searching for it again to call load.

Could you comment on this? If its to keep the API clean as all the other calls take `displaySetInstanceUID` that is fine.

Reply:
I just removed the comment, I agree it makes the API cleaner, and isn't an efficiency bottleneck as this doesn't happen too often.